### PR TITLE
Add additional matchers to package (nevra) filter

### PR DESCRIPTION
**Pull request by parlt91**
## What does this PR change?

Add additional matchers (lower, lowereq, greater, greatereq) to CLM package (NEVRA) filter

## GUI diff

No difference.

- [x] **DONE**

## Documentation

- No documentation needed: **TODO: Check docs if there are mentions about the package (NEVRA) matchers**

- [x] **DONE**

## Test coverage

- Unit tests were added

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/issues/13617
Tracks https://github.com/uyuni-project/uyuni/issues/3406

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/domain/contentmgmt/FilterCriteria.java [hunk 1]
```diff
@@ -69,7 +69,9 @@
     static {
         validCombinations.add(Triple.of(PACKAGE, CONTAINS, "name"));
         validCombinations.add(Triple.of(PACKAGE, EQUALS, "nevr"));
+        validCombinations.add(Triple.of(PACKAGE, GREATEREQ, "nevr"));
         validCombinations.add(Triple.of(PACKAGE, EQUALS, "nevra"));
+        validCombinations.add(Triple.of(PACKAGE, GREATEREQ, "nevra"));
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/domain/contentmgmt/FilterCriteria.java:74]: Does it make sense to "only" implement "greater equal" ?
Why not just implement all variants in one shot?

Comment (parlt91) [reply java/code/src/com/redhat/rhn/domain/contentmgmt/FilterCriteria.java:74]: Yes that was my plan :)
I am just not satisfied with the way I do the comparison down below yet.
The way we currently save the value (nevra) in the db it's hard to exract the evr part so I might need to do some changes there (still thinking about solutions)

**Review by cbbayburt — COMMENTED** [review]
Looks good :+1: 

I'm just wondering how it would behave if the NEVRA can't be parsed properly. With the existing "contains" filter, it's predictable. But I wonder if there would be some confusing results, especially with the comparison.

Unfortunately, we don't have any kind of validation/exception neither in the UI nor in the `PackageEvr` class.

Maybe we can start by adding some wrong parsing cases in the `PackageEvrTest` and see how it behaves?

File: web/html/src/manager/content-management/shared/business/filters.enum.ts [hunk 2]
```diff
@@ -86,6 +96,11 @@ const filterMatchers: FilterMatcherEnumType = {
     text: t("later or equal"),
     longDescription: t("later or equal than"),
   },
+  GREATER: {
+    key: "greater",
+    text: t("later"),
+    longDescription: t("later than"),
```
Comment (hustodemon) [inline web/html/src/manager/content-management/shared/business/filters.enum.ts:102]: Hm, looking at this, shouldn't this be "greater" and "greater than"? Same goes for the existing text above...

I think we only used this for the patch issue date so far, where "later" made sense, but now it'd be better to change it is more general "greater"/"greater than".

**Review by hustodemon — COMMENTED** [review]
Left just one comment, I also think it would be nice to write some tests (in `ContentFilterTest.java`).

The code LGTM :+1: !

**Review by Etheryte — APPROVED** [review]
Frontend changes look good, thanks! 👍

File: java/code/src/com/redhat/rhn/domain/contentmgmt/PackageFilter.java [hunk 3]
```diff
@@ -79,6 +83,18 @@ public boolean test(Package pack) {
         }
     }
 
+    private static String getEvr(String field, String value) {
+        if (field.equals("nevr")) {
+            return value.replaceAll("(.*)-(.*:)?(.*)-(.*)", "$2$3-$4");
+        }
+        else if (field.equals("nevra")) {
+            return value.replaceAll("(.*)-(.*:)?(.*)-(.*)\\.(.*)", "$2$3-$4");
+        }
```
Comment (hustodemon) [inline java/code/src/com/redhat/rhn/domain/contentmgmt/PackageFilter.java:114]: One last-minute thought I got: does this play well with the Debian versions? I think it's worth checking it and adding it to the unit tests.

**Review by hustodemon — COMMENTED** [review]
LGTM, but before my approving, please check and answer my only comment.

File: java/code/src/com/redhat/rhn/domain/contentmgmt/PackageFilter.java (hunk 3 above)
Comment (parlt91) [reply java/code/src/com/redhat/rhn/domain/contentmgmt/PackageFilter.java:114]: Added some test for debian versioning it seems to work fine.

**Review by hustodemon — APPROVED** [review]
LGTM :+1:

**Comment by parlt91** [comment]
@cbbayburt Could you have another look on the PR?

**Review by cbbayburt — APPROVED** [review]

**Pull request merged**